todo_app/data/api.py

```diff
@@ -0,0 +1,114 @@
+        arguments = {'fields': 'name', 'lists': 'open'}
+        response = requests.get(boards_url, params=key_and_token, data=arguments)
+        json_response = response.json()
+        return json_response
+    
+    def create_board(self, name):
+        name = name 
+        url = "https://api.trello.com/1/boards/"
+        key_and_token = self.key_and_token
+        arguments = { 'name': name}
+        response = requests.request( "POST", url, params=key_and_token, data=arguments)
+        json_response = response.json()
+
+        print(json_response)
+        return json_response
+    
+    def delete_board(self, board_id):
+        board_id = board_id 
+        url = "https://api.trello.com/1/boards/" + board_id
+        key_and_token = self.key_and_token
+        response = requests.delete(url, params=key_and_token)
+       
+    def create_list(self, name, board_id):
+        url = "https://api.trello.com/1/lists"
+        key_and_token = self.key_and_token
+        arguments = { 'name': name , 'idBoard': board_id}
+        response = requests.request( "POST", url, params=key_and_token, data=arguments)
+        json_response = response.json()
+
+        print(json_response)
+        return json_response
+        
+    def get_lists(self): 
+        print("get lists board id:")
+        print(self.boardId)
+        boardId = self.boardId
+        lists_url = self.TRELLO_URL + '/boards/' + boardId + '/lists'
+        headers = {"Accept": "application/json"}
+        key_and_token = self.key_and_token
+        arguments = {'fields': 'name', 'lists': 'open' }
+        response = requests.get(lists_url, params=key_and_token, data=arguments)
+        lists = response.json()
+        print(lists)
+        return lists
+
+    def get_cards_for_lists(self, list_id):
+        cards_url = self.TRELLO_URL + '/lists/' + list_id + '/cards/'
+        key_and_token = self.key_and_token
+        response = requests.get(cards_url, params=key_and_token)
+        json_response = response.json()
+        print(json_response)
+        return list(map(Card, json_response))
+
+    def add_item(self, title):
+        url = self.TRELLO_URL + '/cards'
+        id_list = self.to_do_id
+        key_and_token = self.key_and_token
+        name = title
+        description = 'I made this card using the Trello API :fist:'
+        arguments = {'name': name,  'desc': description, 'idList' : id_list} 
+        response = requests.post(url, params=key_and_token, data=arguments)  
+        
+    def move_to_in_progress(self, card_id):
+        id_list = self.doing_id
+        url = self.TRELLO_URL + '/cards/' + card_id + '?idList=' + id_list
+
+        key_and_token = self.key_and_token
+        arguments = {'idList' : id_list}
+        response = requests.put(url, params=key_and_token, data=arguments) 
+    
+    def move_to_done(self, card_id):
+        id_list = self.done_id
+        url = self.TRELLO_URL + '/cards/' + card_id + '?idList=' + id_list
```

Comment:
Here we specify the `idList` field in both the arguments and the query parameters. We should can remove the query parameter, and just rely on the arguments (and similar in the previous method)

Reply:
Done